Sources/Core/AWSClientRuntime/HttpContextBuilder+Extension.swift

```diff
@@ -56,6 +52,16 @@ extension HttpContext {
         // setup client to server
         let messageEncoder = AWSClientRuntime.AWSEventStream.AWSMessageEncoder()
         let messageSigner = AWSClientRuntime.AWSEventStream.AWSMessageSigner(encoder: messageEncoder) {
+            guard let authScheme = self.getSelectedAuthScheme() else {
+                throw ClientError.authError(
+                    "Signer for event stream could not be loaded because auth scheme was not configured."
+                )
+            }
+            guard let signer = authScheme.signer else {
+                throw ClientError.authError("Signer was not configured for the selected auth scheme.")
+            }
+            return signer
```

Comment:
this should all be in a helper function as the return statement makes this function confusing without carefully inspecting where inner scope begins and ends + consider ways to make error messages more direct as it is currently difficult to distinguish the errors / would be difficult to read them out of context

Reply:
Changed structure as recommended.